monitoring/queuemon.go

```diff
@@ -0,0 +1,207 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package monitoring
+
+import (
+	"fmt"
+	"time"
+
+	outqueue "github.com/elastic/beats/v7/libbeat/publisher/queue"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/opt"
+
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter"
+
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter/expvar"
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter/log"
+)
+
+//QueueMonitor is the main handler object for the queue monitor, and will be responsible for startup, shutdown, handling config, and persistent tracking of metrics.
+type QueueMonitor struct {
+	// An array of user-configured outputs
+	outputs []reporter.Reporter
+	//user-configured reporting interval
+	interval time.Duration
+	done     chan struct{}
+	// handler for the event queue
+	queue outqueue.Queue
+	log   *logp.Logger
+	// A awkward no-op if a user has disabled monitoring
+	bypass bool
+
+	// Count of times the queue has reached a configured limit.
+	queueLimitCount uint64
+}
+
+//Config is the intermediate struct representation of the queue monitor config
+type Config struct {
+	Outputs  ReporterConfig `config:"outputs"`
```

Comment:
```suggestion
	Reporters  ReporterConfig `config:"outputs"`
```

There are a few places where output is still used as the name.